incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -31,32 +37,85 @@ var _ = Describe("HNCConfiguration", func() {
 		}).Should(Succeed())
 	})
 
-	It("should propagate objects whose types have been added to HNCConfiguration", func() {
+	It("should set mode of Roles and RoleBindings as propagate by default", func() {
+		config := getHNCConfig(ctx)
+
+		Eventually(hasTypeWithMode("rbac.authorization.k8s.io/v1", "Role", api.Propagate, config)).Should(BeTrue())
+		Eventually(hasTypeWithMode("rbac.authorization.k8s.io/v1", "RoleBinding", api.Propagate, config)).Should(BeTrue())
+	})
+
+	It("should propagate Roles by default", func() {
 		setParent(ctx, barName, fooName)
-		makeSecret(ctx, fooName, "foo-sec")
 
-		// Wait 1 second to give "foo-sec" a chance to be propagated to bar, if it can be propagated.
-		time.Sleep(1 * time.Second)
-		// Foo should have "foo-sec" since we created it there.
-		Eventually(hasSecret(ctx, fooName, "foo-sec")).Should(BeTrue())
-		// "foo-sec" is not propagated to bar because Secret hasn't been configured in HNCConfiguration.
-		Eventually(hasSecret(ctx, barName, "foo-sec")).Should(BeFalse())
+		Eventually(hasRole(ctx, barName, "foo-role")).Should(BeTrue())
+		Expect(roleInheritedFrom(ctx, barName, "foo-role")).Should(Equal(fooName))
+	})
+
+	It("should propagate RoleBindings by default", func() {
+		setParent(ctx, barName, fooName)
 
+		Eventually(hasRoleBinding(ctx, barName, "foo-role-binding")).Should(BeTrue())
+		Expect(roleBindingInheritedFrom(ctx, barName, "foo-role-binding")).Should(Equal(fooName))
+	})
+
+	It("should set CritSingletonNameInvalid condition if singleton name is wrong", func() {
+		nm := "wrong-config-1"
 		Eventually(func() error {
-			c := getHNCConfig(ctx)
-			return addSecretToHNCConfig(ctx, c)
+			config := &api.HNCConfiguration{}
+			config.ObjectMeta.Name = nm
+			return updateHNCConfig(ctx, config)
 		}).Should(Succeed())
 
+		Eventually(hasHNCConfigurationConditionWithName(ctx, api.CritSingletonNameInvalid, nm)).Should(BeTrue())
+	})
+
+	It("should set ObjectReconcilerCreationFailed condition if an object reconciler creation fails", func() {
+		// API version of Secret should be "v1"
+		addToHNCConfig(ctx, "v2", "ConfigMap", api.Propagate)
+
+		Eventually(hasHNCConfigurationCondition(ctx, api.ObjectReconcilerCreationFailed)).Should(BeTrue())
+	})
+
+	It("should unset ObjectReconcilerCreationFailed condition if an object reconciler creation later succeeds", func() {
+		// API version of LimitRange should be "v1"
+		addToHNCConfig(ctx, "v2", "LimitRange", api.Propagate)
+
+		Eventually(hasHNCConfigurationCondition(ctx, api.ObjectReconcilerCreationFailed)).Should(BeTrue())
+
+		updateHNCConfigSpec(ctx, "v2", "v1", "LimitRange", "LimitRange", api.Propagate, api.Propagate)
+
+		Eventually(hasHNCConfigurationCondition(ctx, api.ObjectReconcilerCreationFailed)).Should(BeFalse())
+	})
```

Comment:
Maybe add that the type matches in addition to the error code? I saw your discussions on if to add the type to the struct. Once you decide, maybe add the type match here either by matching string or type subfield.

Reply:
I will add the match for the message for now. :) Thanks for the suggestion!